radicle-node/src/worker.rs

```diff
@@ -403,6 +355,90 @@ impl<G: Signer + Ecdh + 'static> Worker<G> {
             }
         }
     }
+
+    fn tunnel_fetch<Specs>(
+        &self,
+        repo: &fetch::StagedRepository,
+        specs: Specs,
+        session: WireSession<G>,
+    ) -> (WireSession<G>, Result<(), FetchError>)
+    where
+        Specs: fetch::AsRefspecs,
+    {
+        let mut tunnel = match Tunnel::with(session, net::SocketAddr::from(([0, 0, 0, 0], 0))) {
+            Ok(tunnel) => tunnel,
+            Err((session, err)) => return (session, Err(err.into())),
+        };
+        let res = self._fetch(repo, specs, &mut tunnel);
+        let mut session = tunnel.into_session();
+        log::debug!(target: "worker", "Sending `done` packet to remote..");
+        if let Err(err) = pktline::done(&mut session) {
+            log::error!(target: "worker", "Fetch error: error sending `done` packet: {err}");
+        }
+        (session, res)
+    }
+
+    fn _fetch<Specs>(
```

Comment:
```suggestion
    fn _fetch<S>(
```
I tend to prefer this because when reading the function parameters it makes it clearer what is a generic vs. concrete type.

Reply:
Sure :+1: